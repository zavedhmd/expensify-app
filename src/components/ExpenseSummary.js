import React from 'react';
import expenseTotal from '../selectors/expense-total';
import selectExpenses from '../selectors/expenses';
import { connect } from "react-redux";
import numeral from 'numeral'

export const ExpenseSummary = ({expenseCount, expenseTotal}) => (
    <div>
        <h2>{`Viewing total amount : ${numeral(expenseTotal / 100).format('$0,00.00')}
             and with : ${expenseCount} expenses`}
        </h2>
    </div>
)

const matchStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount : visibleExpenses.length,
        expenseTotal : expenseTotal(visibleExpenses)
    }
}

export default connect(matchStateToProps)(ExpenseSummary);