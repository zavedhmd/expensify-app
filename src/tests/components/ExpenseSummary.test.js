import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should match snapshot', () => {
    const wrapper = shallow(<ExpenseSummary expenses={1} totalAmount={5}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should update prop', () => {
    const wrapper = shallow(<ExpenseSummary expenses={23} totalAmount={100}/>);
    expect(wrapper).toMatchSnapshot();
})