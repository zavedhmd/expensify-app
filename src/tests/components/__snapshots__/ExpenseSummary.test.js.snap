// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <h2>
    Viewing total amount : $0.00
                 and with : undefined expenses
  </h2>
</div>
`;

exports[`should update prop 1`] = `
<div>
  <h2>
    Viewing total amount : $0.00
                 and with : undefined expenses
  </h2>
</div>
`;
